### PR TITLE
Revert "Moved subscribe and filter into network controller (#16693)"

### LINKED ISSUE #17467: [Bug]: MetaMask stays loading indefinetly and get error `StreamProvider - Unknown response id for response: [object Object]` when doing *some* contract interactions
### Describe the bug

**Problem**: when I call the function for mint an ERC1155 token, MetaMask stays loading indefinetly and I get the console error `Error in event handler: Error: StreamProvider - Unknown response id for response: [object Object]` .

https://user-images.githubusercontent.com/54408225/215133926-378001fe-6c51-45d9-8e64-930da22e90a6.mp4

This also happens when I call another contract function.

https://user-images.githubusercontent.com/54408225/215423675-bcd1b1e4-cc28-4e0a-9e29-911371eb2eab.mp4

[EDITED] It seems the error could start whenever we deploy a contract. It's not 100% deterministic, so further investigation needs to be done.

### Steps to reproduce [EDITED]
1. Go to Remix https://remix.ethereum.org/
2. Create a contract
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.17;

contract Fallback {
    event Log(string func, uint gas);
    fallback() external payable {
        emit Log("fallback", gasleft());
    }
    receive() external payable {
        emit Log("receive", gasleft());
    }
    function getBalance() public view returns (uint) {
        return address(this).balance;
    }
}
```
3. Deploy this contract on Goerli - see error on console // if not, deploy another contract
4. Create another contract 
```
contract SendToFallback {
    uint val = 0;
    function transferToFallback(address payable _to) public payable {
        _to.transfer(msg.value);
    }

    function callFallback(address payable _to) public payable {
        (bool sent, ) = _to.call{value: msg.value}("");
        require(sent, "Failed to send Ether");
    }
    function count(uint _val) public {
        val = _val;
    }
}
```
6. Deploy this contract on Goerli - see error on console

After this, we are into a broken state, shown in the videos above.

### Error messages or log output

_No response_

### Version

develop [EDITED]

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gudahtt**
This reverts commit 6f6984fa58c4797e809387b3b3d0b28e72b252bf. That commit was an RPC middleware refactor intended to move the subscribe and filter middleware into the network controller, to simplify the process of sharing this middleware between clients.

This refactor resulted in `eth_subscribe` notifications being sent on the wrong connections, causing the UI to break in some cases (the UI `provider` connection does not support notifications). This happened because the `setupProviderEngine` function runs per-connection, whereas the engine setup inside the network controller is global. The global network client cannot support notifications because it has no way to route them; they'll need to stay in the per-connection provider engine.

Closes #17467

## Manual Testing Steps

* Attempt to reproduce #17467
* Test that the filter and subscribe methods work correctly
  * I'd recommend running this branch of the `test-dapp` locally to help with testing: https://github.com/MetaMask/test-dapp/pull/202

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by Gudahtt** [comment]
>Sufficient automated test coverage has been added

We should be able to cover this with an e2e test. I'll have to give that some thought. I don't think it'd be feasible to capture this error in a unit test, since it's more of an integration problem.

**Comment by seaona** [comment]
Looks like this fixed it. I cannot reproduce the error anymore @Gudahtt

File: test/e2e/tests/eth-subscribe.spec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        await driver.executeScript(setupSubscriptionListener);
+        await driver.executeScript(`
+          window.ethereum.request({
+            method: 'eth_subscribe',
+            params: ['newHeads']
+          });
+        `);
+
+        // Verify that new block is seen on first dapp
```
Comment (Gudahtt) [inline test/e2e/tests/eth-subscribe.spec.js:60]: We use a default block time of 2 in our Ganache setup. Our `findElement` timeout is more than long enough to wait for this, but in the future we can improve this test by explicitly mining a block here using the `evm_mine` method enabled by this PR: https://github.com/MetaMask/metamask-extension/pull/18149

File: test/e2e/tests/eth-subscribe.spec.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+
+        const setupSubscriptionListener = `
+          const responseContainer = document.createElement('div');
```
Comment (Gudahtt) [inline test/e2e/tests/eth-subscribe.spec.js:34]: We can replace this with the built-in test dapp subscribe functionality after this PR is merged: https://github.com/MetaMask/test-dapp/pull/202

File: test/e2e/tests/eth-subscribe.spec.js (hunk 1 above)
Comment (adonesky1) [inline test/e2e/tests/eth-subscribe.spec.js:60]: by new block do you mean the div that is appended when the subscription response comes back? Or do you mean a block like an ethereum block?  If its the former, which it seems like it is maybe we should use a different word than "block"?

Comment (Gudahtt) [reply test/e2e/tests/eth-subscribe.spec.js:60]: I meant the latter, an Ethereum block. `newHeads` will emit a notification for each new block

Comment (adonesky1) [reply test/e2e/tests/eth-subscribe.spec.js:60]: Ok gotcha.

**Comment by adonesky1** [comment]
> Test that the filter and subscribe methods work correctly
> I'd recommend running this branch of the test-dapp locally to help with testing: https://github.com/MetaMask/test- 
> dapp/pull/202

I tried this both on this branch and on `develop` and it worked on both...

**Review by adonesky1 — DISMISSED** [review]
LGTM! Tested and verified that `eth_subscribe` subscription messages are no longer broadcast to all pages where we inject our provider

File: test/e2e/tests/eth-subscribe.spec.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          });
+        `);
+
+        // Verify that the new block is seen on the first dapp
+        await driver.findElement('[data-testid="eth-subscribe-response"]');
+
+        // Switch to the second dapp
+        await driver.openNewPage('http://127.0.0.1:8081/');
+
+        // Setup the same subscrption listener as on the first dapp, but without registering a new subscription
```
Comment (adonesky1) [inline test/e2e/tests/eth-subscribe.spec.js:68]: nit
```suggestion
        // Setup the same subscription listener as on the first dapp, but without registering a new subscription
```

File: test/e2e/tests/eth-subscribe.spec.js (hunk 1 above)
Comment (Gudahtt) [reply test/e2e/tests/eth-subscribe.spec.js]: I've just updated these comments to be a bit clearer: https://github.com/MetaMask/metamask-extension/pull/18129/commits/2aa96dac4a52f2d94a0bdf7a74b10c728bb8bea9

File: test/e2e/tests/eth-subscribe.spec.js (hunk 3 above)
Comment (Gudahtt) [reply test/e2e/tests/eth-subscribe.spec.js]: Thanks! Fixed here: https://github.com/MetaMask/metamask-extension/pull/18129/commits/fde1a64b7b93497648c1ecdc28d01f7ddb63fe0b

**Review by adonesky1 — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM, thanks!

**Pull request merged**